gc/base/MemoryPoolLargeObjects.cpp

```diff
@@ -466,17 +448,7 @@ MM_MemoryPoolLargeObjects::resetLOASize(MM_EnvironmentBase* env, double newLOARa
 
 		uintptr_t resizeSize = 0;
 		/* Does this leave a reasonable sized LOA ? */
-		if (!isSizeEnoughForLOA(env, newLOASize)) {
-			/* No.. make LOA empty as not even big enough for one free chunk */
-			_currentLOARatio = 0;
-			_soaSize = oldAreaSize;
-			_loaSize = 0;
-		} else {
-			_currentLOARatio = newLOARatio;
-			_loaSize = newLOASize;
-			/* ..and SOA is whats left after LOA allocation */
-			_soaSize = oldAreaSize - _loaSize;
-		}
+		isSizeEnoughForLOA(env, oldAreaSize, newLOASize, newLOARatio);
```

Comment:
Move declaration of newLOASize and oldAreaSize down one block to where is initialized first time.